libminifi/include/utils/Averager.h

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <vector>
+#include <mutex>
+
+namespace org::apache::nifi::minifi::utils {
+
+template<typename T>
+concept Summable = requires(T x) { x + x; };  // NOLINT(readability/braces)
+
+template<typename T>
+concept DividableByInteger = requires(T x, uint32_t divisor) { x / divisor; };  // NOLINT(readability/braces)
+
+template<typename ValueType>
+requires Summable<ValueType> && DividableByInteger<ValueType>
+class Averager {
```

Comment:
We should discuss it with @fgerlits too as it was his comment to extract it to a generic utility. I thought it was not that specific to this use case and could be reused later as it can calculate the average of any numerical or time values. I think the `Averager` together with the `sample_size` constructor argument should be descriptive enough to use it, but I'm open for discussing this on a team level.

Reply:
Yes, let's discuss this in person.  I think separating `Averager` further into a circular buffer and an averaging function which works on the buffer is a good further refactoring idea, but it is also fairly generic and reusable as it is.

Reply:
Moved `Averager` closer to usage in 0bc6bda36d41dd7582bd0780220d2ff4303e50e8